### ISSUE TITLE
why big size bitmap could not display correctly

### ISSUE THREAD

**Issue by runweok** [facebook/fresco#606]
fresco version: 0.7.0

sample image:(pixel : 431X5037)
http://matocloud.com.iloveoops.com/ODE1MjkxNDQxOTM4NjgyNTA0.jpg?op=imageView2&mode=3&width=720

problem:
the image always be fuzzy。
how can i get the image clearly? even if show a part of it.

xml:
```
<com.facebook.drawee.view.SimpleDraweeView xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:fresco="http://schemas.android.com/apk/res-auto"
    android:id="@+id/my_image_view"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    fresco:actualImageScaleType="center"
    fresco:fadeDuration="300"
    fresco:placeholderImageScaleType="fitCenter"
    fresco:progressBarAutoRotateInterval="1000"
    fresco:progressBarImage="@drawable/global_tick_loading"
    fresco:progressBarImageScaleType="centerInside" />

```
code:
```
ImageRequestBuilder requestBuilder = ImageRequestBuilder.newBuilderWithSource(url)
               .setAutoRotateEnabled(true)
                    .setImageType(ImageRequest.ImageType.DEFAULT);
            ImageRequest request = requestBuilder.build();
            PipelineDraweeController controller = (PipelineDraweeController) Fresco.newDraweeControllerBuilder()
                    .setOldController(sdv.getController())
                    .setImageRequest(request)
                    .setAutoPlayAnimations(true)
                    .build();
            sdv.setController(controller);
```

**Comment by IanChilds** [comment]
What is the width of your layout? 431 is quite small so perhaps the image is having to be upscaled horizontally?

**Comment by wstczlt** [comment]
I have also met this problem when showing these long height images, because the image is very long (larger than OpenGL render height), you must use software layer. This is a mot prefect  solution:
 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
                if (largeImage.width > 2048 || largeImage.height > 2048) {
                    holder.mImageView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
                } else {
                    holder.mImageView.setLayerType(View.LAYER_TYPE_HARDWARE, null);
                }
            }

You can have a try.

**Comment by runweok** [comment]
to IanChilds:
   That's the question.All of fresco  ScaleType: center, centerCrop,focusCrop...resize the image,maybe result in fuzzy.
to wstczlt:
  its a net url image, i can't know the real image size, it's seem to not work .

**Comment by IanChilds** [comment]
Android has a limit of 4k * 4k on images, so the height of your image is too great. You need to resize the image to make sure it fits within Android's limits.

**Comment by caoyunxuan07** [comment]
why my a4 image has been resized to half width and height, which is obviously under 4k * 4k limit.
origin size(a4):2479x3508
get from fresco: 1240x1754
```
ImageRequestBuilder.newBuilderWithResourceId(R.drawable.default_blank_a4)
                    .setProgressiveRenderingEnabled(true)
                    .build()
```

**Comment by lambdapioneer** [comment]
Hi @caoyunxuan07,

The default `MAX_BITMAP_SIZE` that Fresco is using is 2048px, as this is the limit that some older devices can process.

The pull-request https://github.com/facebook/fresco/pull/1461 (thanks @kunny) has made this limit configurable such that you can override it for a image request. This will be included in the next release 0.14.0 that we hope to ship very soon.
